services/queue/src/main.js

```diff
@@ -33,6 +33,10 @@ const DEFAULT_UPDATE_FREQUENCY = '30 minutes';
 // for supported bulk deletion S3 operations
 const MAX_BULK_DELETE_SIZE = 1000;
 
+// maximum number of records to process at once in claim, deadline, and dependency resolvers
+// this is to limit total amount of concurrent updates to DB
+const NUMBER_OF_RECORDS_TO_PROCESS = 32;
```

Comment:
I'm wondering if we should make this configurable and default to 32 so that we can have finer control over it without having to make additional code changes. Wdyt?

Reply:
Haha, this was initially in `config.yml` :) But it's more of a node<>pg library that has own limits on number of concurrent connections. And after Dustin's comments I moved it back to constant. I'll keep it as is, no need to modify on deployment level